lib/ranked-model/ranker.rb

```diff
@@ -112,9 +112,12 @@ def position_at value
         update_index_from_position
       end
 
-      def rank_at value
+      def rank_at value, rebalance: false
```

Comment:
I don't think this boolean for `rebalance` is necessary since looking at line 243, we're going to be setting an `Integer` so `position.is_a?(Integer)` will satisfy the `unless` already.

Reply:
Sorry, I see you want to skip this in the case of a rebalance. Can you let me know why? :)

Reply:
I saw some tests fail when rebalancing occurred here because `position` would be a symbol (e.g. `:down`), but the `rank_at` method would get called twice. First to rebalance the ranks, and then again to set the new value.

The result was that `position` was being set on the rebalance pass to the _current_ relative rank, and then the next call that would update the position would set the `rank_at` incorrectly.

By using this boolean for internal rebalancing calls to `rank_at`, the position change was persisted until the final rank was set.

Reply:
I'm not a huge fan of the boolean flag, so if you have any other suggestion to handle that case, I would love to take it.

Reply:
I've had a deeper look at this. I think rather than rely on `rank_at` when rebalancing, we should just mutate the column directly. This removes the need for the boolean. What do you think?

Reply:
Makes sense to me!